### PR TITLE
Stop using the k8s.io/utils module directly

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [x] Have you added automated tests?

**Type of Changes:**

 
 - [ ] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [ ] Testing enhancement
 - [x] Other

### CODE REVIEW

File: internal/controller/pgupgrade/utils.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func (r *PGUpgradeReconciler) setControllerReference(
 			Kind:               "PGUpgrade",
 			Name:               owner.GetName(),
 			UID:                owner.GetUID(),
-			BlockOwnerDeletion: pointer.BoolPtr(true),
-			Controller:         pointer.BoolPtr(true),
+			BlockOwnerDeletion: initialize.Pointer(true),
```
Comment (benjaminjb) [inline internal/controller/pgupgrade/utils.go:52]: 🤔 would this complain... if it wasn't clear that we're returning a pointer to a bool?

File: internal/initialize/primitives.go [hunk 2]
```diff
@@ -25,12 +25,25 @@ func ByteMap(m *map[string][]byte) {
...
+}
+
 // Int32 returns a pointer to v.
 func Int32(v int32) *int32 { return &v }
 
 // Int64 returns a pointer to v.
 func Int64(v int64) *int64 { return &v }
 
+// Pointer returns a pointer to v.
+func Pointer[T any](v T) *T { return &v }
```
Comment (benjaminjb) [inline internal/initialize/primitives.go:45]: 🤔 Def not suggesting this now, but any reason to keep around non-generic funcs that do this same thing?

**Review by benjaminjb — APPROVED** [review]
Just some reminders for myself to go read up on generics

File: internal/initialize/primitives.go (hunk 2 above)
Comment (cbandy) [reply internal/initialize/primitives.go:45]: The functions for numeric types might be convenient with untyped constants; `initialize.Int32(0)` versus `initialize.Pointer(int32(0))`. The latter, "initialize pointer int32," sounds more clear to me at this moment.

https://go.dev/ref/spec#Constants
> The default type of an untyped constant is `bool`, `rune`, `int`, `float64`, `complex128` or `string` respectively, depending on whether it is a boolean, rune, integer, floating-point, complex, or string constant.

File: internal/controller/pgupgrade/utils.go (hunk 1 above)
Comment (cbandy) [reply internal/controller/pgupgrade/utils.go:52]: Yes. The types in the struct are... established. When I change `true` to `9`, `go build` and `go test` tell me:

```
internal/controller/pgupgrade/utils.go:52:24: cannot use initialize.Pointer(9) (value of type *int) as type *bool in struct literal
```

**Pull request merged**